src/extra/libs/rgbdec/lv_rgb_dec.c

```diff
@@ -0,0 +1,523 @@
+/**
+ * @file lv_rgb_dec.c
+ *
+ */
+
+/*********************
+ *      INCLUDES
+ *********************/
+
+#include "../../../lvgl.h"
+
+
+/*********************
+ *      DEFINES
+ *********************/
+#define CF_BUILT_IN_FIRST LV_IMG_CF_TRUE_COLOR
+#define CF_BUILT_IN_LAST LV_IMG_CF_ALPHA_8BIT
+
+/**********************
+ *      TYPEDEFS
+ **********************/
+
+typedef struct {
+    lv_fs_file_t f;
+    lv_color_t * palette;
+    lv_opa_t  *  opa;
+} lv_img_decoder_built_in_data_t;
+
+
+/**********************
+ *  STATIC PROTOTYPES
+ **********************/
+static lv_res_t lv_img_decoder_built_in_line_true_color(lv_img_dec_dsc_t * dsc, lv_coord_t x, lv_coord_t y,
+                                                        lv_coord_t len, uint8_t * buf);
+static lv_res_t lv_img_decoder_built_in_line_alpha(lv_img_dec_dsc_t * dsc, lv_coord_t x, lv_coord_t y,
+                                                   lv_coord_t len, uint8_t * buf);
+static lv_res_t lv_img_decoder_built_in_line_indexed(lv_img_dec_dsc_t * dsc, lv_coord_t x, lv_coord_t y,
+                                                     lv_coord_t len, uint8_t * buf);
+
+static lv_res_t lv_img_decoder_built_in_accept(const lv_img_src_t * src, uint8_t * caps);
+
+static lv_res_t lv_img_decoder_built_in_open(lv_img_dec_dsc_t * dsc, const lv_img_dec_flags_t flags);
+
+static lv_res_t lv_img_decoder_built_in_read_line(lv_img_dec_dsc_t * dsc,
+                                                  lv_coord_t x, lv_coord_t y, lv_coord_t len, uint8_t * buf);
+
+static void lv_img_decoder_built_in_close(lv_img_dec_dsc_t * dsc);
+
+void lv_bin_init(void);
+
+
+void lv_bin_init()
```

Comment:
`bin` is too generic and `lv_rgb_dec_c` is misleading as alpha only and indexed images are also decoded here.  
What about having 3 decoders here:
- `lv_imgdec_rgb_init()`
- `lv_imgdec_indexed_init()`
- `lv_imgdec_alphaonly_init()`

Reply:
The issue is how to distinguish between those ?
Right now, this works because it's the last decoder to try and it'll accept any format that has a `img_dsc` header first in its data. Yet, this is very fragile since this header is not very specific and can accept anything raw binary as long as a zero appears at the right byte position. 

Hopefully, since it's tried last, another decoder is more likely to accept the data so we are safe in the most case. But as soon as you consider that we'll use LVGL img's feature on user controllable data (like pictures downloaded from the internet ;-), then it's game over. The only thing a user has to do is set a very huge width or height in the header and it'll lead to segmentation fault.

It's mainly used for the embedded raw image decoder so maybe `lv_raw_imgdec` would be better ? What do you think ?

Reply:
> The issue is how to distinguish between those ?

True.

> It's mainly used for the embedded raw image decoder so maybe lv_raw_imgdec would be better ? What do you think ?

`lv_rawimg_dec`?

Reply:
Hail to `lv_rawimg_dec` then.